src/Orleans/Async/BatchWorker.cs

```diff
@@ -86,8 +86,8 @@ private void CheckForMoreWork()
             }
 
             // to be safe, must do the signalling out here so it is not under the lock
-            if (signalThunk != null)
-                signalThunk();
+            if (signal != null)
+                signal.SetResult(task_to_signal);
```

Comment:
super nit picking (only if you tackle the other one): no need to have an `if`, just do `signal?.SetResult(taskToSignal);`. In this particular case there can be no race between the if check and the method call, but it's a good practice in general.